org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/validation/ProjectAwareUniqueClassNameValidator.xtend

```diff
@@ -174,15 +174,17 @@ class ProjectAwareUniqueClassNameValidator extends UniqueClassNameValidator {
 
 	def protected isDerived(IResource resource) {
 		try {
-			if (derivedResourceMarkers.findDerivedResourceMarkers(resource).length >= 1) {
+			if (derivedResourceMarkers !== null && derivedResourceMarkers.findDerivedResourceMarkers(resource).length >= 1) {
 				return true
 			}
 			val outputConfigurations = context.get(OUTPUT_CONFIGS) as Collection<OutputConfiguration>
 			if (outputConfigurations !== null) {
 				val projectRelativePath = resource.projectRelativePath
 				for(outputConfiguration: outputConfigurations) {
 					for(dir: outputConfiguration.outputDirectories) {
-						if (new Path(dir).isPrefixOf(projectRelativePath)) {
+						val sourceMappingForOutput = outputConfiguration.sourceMappings.filter[it.sourceFolder.endsWith(dir)].head
```

Comment:
I've to admit that I find this logic hard to understand. Could you please extract one or the other method and give it a name?

Reply:
Will do